source/extensions/quic_listeners/quiche/platform/quic_file_utils_impl.cc

```diff
@@ -0,0 +1,55 @@
+// NOLINT(namespace-envoy)
+
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+#include "extensions/quic_listeners/quiche/platform/quic_file_utils_impl.h"
+
+#include "common/filesystem/directory.h"
+#include "common/filesystem/filesystem_impl.h"
+
+#include "absl/strings/str_cat.h"
+
+namespace quic {
+namespace {
+
+void DFTraverseDirectory(const std::string& dirname, std::vector<std::string>& files) {
+  Envoy::Filesystem::Directory directory(dirname);
+  for (const Envoy::Filesystem::DirectoryEntry& entry : directory) {
+    switch (entry.type_) {
+    case Envoy::Filesystem::FileType::Regular:
+      files.push_back(absl::StrCat(dirname, "/", entry.name_));
+      break;
+    case Envoy::Filesystem::FileType::Directory:
+      if (entry.name_ != "." && entry.name_ != "..") {
+        DFTraverseDirectory(absl::StrCat(dirname, "/", entry.name_), files);
+      }
+      break;
+    default:
+      ASSERT(false,
+             absl::StrCat("Unknow file entry type ", entry.type_, " under directory ", dirname));
+    }
+  }
+}
+
+} // namespace
+
+// Traverses the directory |dirname| and returns all of the files it contains.
+std::vector<std::string> ReadFileContentsImpl(const std::string& dirname) {
```

Comment:
Comment for "upstream", I think this is badly named as it's not reading the file contents.
I don't think you can fix that here though :-/